### PR TITLE
deps(selenium-webdriver): upgrade to selenium 3

### PR DESCRIPTION
**Pull request by cnishina**

### CODE REVIEW

File: lib/element.ts [hunk 1]
```diff
@@ -13,8 +13,8 @@ let logger = new Logger('element');
 
 let WEB_ELEMENT_FUNCTIONS = [
   'click', 'sendKeys', 'getTagName', 'getCssValue', 'getAttribute', 'getText', 'getSize',
-  'getLocation', 'isEnabled', 'isSelected', 'submit', 'clear', 'isDisplayed', 'getOuterHtml',
-  'getInnerHtml', 'getId', 'getRawId', 'serialize', 'takeScreenshot'
+  'getLocation', 'isEnabled', 'isSelected', 'submit', 'clear', 'isDisplayed', 'getId', 'getRawId',
```
Comment (juliemr) [inline lib/element.ts]: we should remove `getRawId` too

File: spec/basic/elements_spec.js [hunk 2]
```diff
@@ -470,15 +470,13 @@ describe('ElementArrayFinder', function() {
     browser.get('index.html#/form');
     var labels = element.all(by.css('#animals ul li')).map(function(elm) {
       return {
-        text: elm.getText(),
-        inner: elm.getInnerHtml()
+        text: elm.getText()
```
Comment (juliemr) [inline spec/basic/elements_spec.js]: We should have something else here to make sure multiple promises are waited for. How about `elm.getTagName`?

File: spec/dependencyTest/seleniumWebdriver_spec.js [hunk 3]
```diff
@@ -35,75 +35,71 @@ var CHROME = {
...
-      'describe', 'it', 'iit']
+  instanceFunctions: ['after', 'afterEach', 'before', 'beforeEach', 'it', 'xit']
 };
 
 describe('selenium-webdriver dependency', function() {
   describe('require("selenium-webdriver").WebDriver', function() {
-    for (var pos1 in WEBDRIVER.staticFunctions) {
-      var staticFunc = WEBDRIVER.staticFunctions[pos1];
-      it('should have a ' + staticFunc + ' function', function() {
+    it('should have static function', function() {
```
Comment (juliemr) [inline spec/dependencyTest/seleniumWebdriver_spec.js]: 'should have static functions'? (and below)

File: spec/mocha/lib_spec.js [hunk 4]
```diff
@@ -42,7 +42,7 @@ describe('protractor library', function() {
       browser.get('index.html').then(function() { finished = true; });
     });
 
-    after('verify mocha waited', function() {
+    after(function() {
```
Comment (juliemr) [inline spec/mocha/lib_spec.js]: is this a change from Mocha? If not, we should continue to support a description for 'after' if it's a valid Mocha construct.

File: lib/element.ts (hunk 1 above)
Comment (cnishina) [inline lib/element.ts]: done.

File: spec/basic/elements_spec.js (hunk 2 above)
Comment (cnishina) [inline spec/basic/elements_spec.js]: Added `elm.getTagName` to the test.

File: spec/mocha/lib_spec.js (hunk 4 above)
Comment (cnishina) [inline spec/mocha/lib_spec.js]: Added back `after('verify mocha waited', `

**Comment by juliemr** [comment]
Please make sure that the description for the big commit links to the selenium-webdriver changelog:  https://github.com/SeleniumHQ/selenium/blob/master/javascript/node/selenium-webdriver/CHANGES.md

and gives a quick heads up about major breaking changes.

File: spec/dependencyTest/seleniumWebdriver_spec.js [hunk 5]
```diff
@@ -3,24 +3,24 @@ var By = require('selenium-webdriver').By;
...
-var Executors = require('selenium-webdriver/executors');
 var SeleniumError = require('selenium-webdriver').error;
 var Remote = require('selenium-webdriver/remote');
 var Testing = require('selenium-webdriver/testing');
 
 var WEBDRIVER = {
   staticFunctions: ['attachToSession', 'createSession'],
+
   instanceFunctions: ['actions', 'wait', 'sleep', 'getCurrentUrl', 'getTitle',
     'takeScreenshot', 'getSession', 'getCapabilities', 'quit', 'touchActions',
```
Comment (juliemr) [inline spec/dependencyTest/seleniumWebdriver_spec.js]: Should we remove 'quit' from here then?

Comment (juliemr) [inline spec/dependencyTest/seleniumWebdriver_spec.js]: Nevermind, wrong place...

Comment (juliemr) [inline spec/dependencyTest/seleniumWebdriver_spec.js]: Great!

**Comment by cnishina** [comment]
:bowtie: :grinning: :blush:

**Pull request merged**